benchmark/collective/resnet/dali.py

```diff
@@ -0,0 +1,271 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+
+import os
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+from nvidia.dali.plugin.paddle import DALIGenericIterator
+
+import paddle
+from paddle import fluid
+
+def convert_data_layout(data_layout):
+    if data_layout=='NCHW':
+        return types.NCHW
+    elif data_layout=='NHWC':
+        return types.NHWC
+    else:
+        assert False, "not supported data_layout:{}".format(data_layout)
+
+class HybridTrainPipe(Pipeline):
```

Comment:
some description is required to help understanding these code, as it is hard to GUESS what this pipeline stands for.

Reply:
It's from the DALI document. Will add some comments.Thanks.

Reply:
Done.